pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -134,7 +134,55 @@ func (c *Cluster) makeDeploymentPVC(m *monConfig) (*v1.PersistentVolumeClaim, er
  * Pod spec
  */
 
-func (c *Cluster) makeMonPod(monConfig *monConfig, hostname string) *v1.Pod {
+func (c *Cluster) setPodAffinity(pod *v1.PodSpec) {
+	pod.NodeSelector = nil
+
+	// try to avoid scheduling monitors within the same available zone
+	//
+	// TODO: the topology key should likely be exposed through the CRD. However,
+	// this does preserve the policy of the current explicit scheduler in Rook
+	// which always looks for the `zone` topology key.
+	pod.Affinity = &v1.Affinity{
+		PodAntiAffinity: &v1.PodAntiAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []v1.WeightedPodAffinityTerm{
+				{
+					Weight: 50,
+					PodAffinityTerm: v1.PodAffinityTerm{
+						TopologyKey: v1.LabelZoneFailureDomain,
+					},
+				},
+			},
+		},
+	}
+
+	// label selector for monitors used in anti-affinity rules
+	monAntiAffinity := v1.PodAffinityTerm{
+		LabelSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				k8sutil.AppAttr: appName,
+			},
+		},
+	}
+
+	// set monitor pod anti-affinity rules. when monitors should never be
+	// co-located (e.g. not AllowMultiplePerHost or HostNetworking) then the
+	// anti-affinity rule is made to be required during scheduling, otherwise it
+	// is merely a preferred policy.
+	paa := pod.Affinity.PodAntiAffinity
+	if c.spec.Mon.AllowMultiplePerNode || c.HostNetwork {
```

Comment:
Should this condition be `!c.HostNetwork`? I think we want RequiredDuringScheduling for the antiaffinity when host networking is enabled.

Reply:
yes, i believe that is correct!